p2p/transport/webrtc/util.go

```diff
@@ -0,0 +1,77 @@
+package libp2pwebrtc
+
+import (
+	"encoding/hex"
+	"math/rand"
+	"strings"
+
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multibase"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/pion/webrtc/v3"
+)
+
+func maFingerprintToSdp(fp string) string {
+	result := ""
+	first := true
+	for pos, char := range fp {
```

Comment:
given it is a function that seems like it could potentially be on some hotpaths I think it is worth considering to improve this algorithm as it seems pretty naive?

Reply:
2 ways to explore:
* using Go's `strings.Builder`
* preallocating a byte slice, doing all the appends in the slice, and then returning `string(slice)`

I have no inituition which one would be better here, only a benchmark would tell. Feel free to commit the benchmark code. Bonus points for adding a comment saying "Using strings.Builder is faster than appending to a string, since it saves allocations" (or whatever the conclusion is).

Reply:
```
goos: linux
goarch: amd64
pkg: github.com/libp2p/go-libp2p/p2p/transport/webrtc
cpu: AMD Ryzen 9 5950X 16-Core Processor            
BenchmarkMaFingerprintToSdp
BenchmarkMaFingerprintToSdp-32            333162              5408 ns/op
BenchmarkIntersperseSDP
BenchmarkIntersperseSDP-32               1692710               749.1 ns/op
BenchmarkIntersperse2
BenchmarkIntersperse2-32                 7304058               151.6 ns/op
PASS
ok      github.com/libp2p/go-libp2p/p2p/transport/webrtc        5.105s
```
IntersperseSDP uses the `strings.Builder` while `intersperse2` uses a byte buffer and is significantly faster.